lib/elixir/lib/code/formatter.ex

```diff
@@ -1337,39 +1315,26 @@ defmodule Code.Formatter do
 
     iodata =
       case Code.Identifier.classify(atom) do
-        type when type in [:callable_local, :callable_operator, :not_callable] ->
-          [?:, string]
-
-        _ ->
-          [?:, ?", String.replace(string, "\"", "\\\""), ?"]
+        type when type in [:callable_local, :callable_operator, :not_callable] -> [?:, string]
+        _ -> [?:, ?", String.replace(string, "\"", "\\\""), ?"]
       end
 
     iodata |> IO.iodata_to_binary() |> string()
   end
 
   defp integer_to_algebra(text) do
     case text do
-      [?0, ?x | rest] ->
-        "0x" <> String.upcase(List.to_string(rest))
-
-      [?0, base | _rest] = digits when base in [?b, ?o] ->
-        List.to_string(digits)
-
-      [?? | _rest] = char ->
-        List.to_string(char)
-
-      decimal ->
-        List.to_string(insert_underscores(decimal))
+      [?0, ?x | rest] -> "0x" <> String.upcase(List.to_string(rest))
+      [?0, base | _rest] = digits when base in [?b, ?o] -> List.to_string(digits)
+      [?? | _rest] = char -> List.to_string(char)
+      decimal -> List.to_string(insert_underscores(decimal))
```

Comment:
Too many clauses grouped together without space.

Reply:
Definitely needs alignment for single-line cases when all fit within the line length bounds, so the above would be:

```elixir
    case text do
      [?0, ?x | rest]                                   -> "0x" <> String.upcase(List.to_string(rest))
      [?0, base | _rest] = digits when base in [?b, ?o] -> List.to_string(digits)
      [?? | _rest] = char                               -> List.to_string(char)
      decimal                                           -> List.to_string(insert_underscores(decimal))
    end
```

Hmm, I wonder how `when`'s would look like when split...  Just testing:

```elixir
    case text do
      [?0, ?x | rest]             -> "0x" <> String.upcase(List.to_string(rest))
      [?0, base | _rest] = digits
        when base in [?b, ?o]     -> List.to_string(digits)
      [?? | _rest] = char         -> List.to_string(char)
      decimal                     -> List.to_string(insert_underscores(decimal))
    end
```